src/base/bittorrent/torrenthandleimpl.cpp

```diff
@@ -200,6 +200,26 @@ TorrentHandleImpl::TorrentHandleImpl(Session *session, const lt::torrent_handle
         if (filesCount() == 1)
             m_hasRootFolder = false;
     }
+
+    // TODO: Remove the following upgrade code in v.4.4
+    const QString spath = actualStorageLocation();
+    for (int i = 0; i < filesCount(); ++i) {
+        const QString filepath = filePath(i);
+        // Move "unwanted" files back to their original folder
+        const QString parentRelPath = Utils::Fs::branchPath(filepath);
+        if (QDir(parentRelPath).dirName() == ".unwanted") {
+            const QString oldName = Utils::Fs::fileName(filepath);
+            const QString newRelPath = Utils::Fs::branchPath(parentRelPath);
+            if (newRelPath.isEmpty())
+                renameFile(i, oldName);
+            else
+                renameFile(i, QDir(newRelPath).filePath(oldName));
+
+            // Remove .unwanted directory if empty
+            qDebug() << "Attempting to remove \".unwanted\" folder at " << QDir(spath + '/' + newRelPath).absoluteFilePath(".unwanted");
+            QDir(spath + '/' + newRelPath).rmdir(".unwanted");
```

Comment:
@thalieht: 

> It doesn't delete the empty .unwanted folder in default save path (created by a torrent without root folder). 

So you are saying that the `.unwanted` directory removal code (which I attached this post to) does not work for torrents without root folders? 

If this is indeed the case, what do you think needs to change to make the deletion execute properly?

Reply:
>So you are saying that the .unwanted directory removal code (which I attached this post to) does not work for torrents without root folders?

I only tested a torrent with the root folder removed (by `[ ] Create subfolder`). No idea if a torrent, natively without one, is treated differently and i can't test now, tomorrow.

>If this is indeed the case, what do you think needs to change to make the deletion execute properly?

Beats me. I'd have to dig deeeep in the methods used here (branchPath, actualStorageLocation etc.) to find out what they do first.

Reply:
>No idea if a torrent, natively without one, is treated differently and i can't test now, tomorrow.

Tested that, same result.